packages/eslint-plugin/tests/rules/non-nullable-type-assertion-style.test.ts

```diff
@@ -61,6 +61,16 @@ const x = 1 as 1;
 declare function foo<T = any>(): T;
 const bar = foo() as number;
     `,
+    `
+function first<T>(array: ArrayLike<T>): T | null {
+  return array.length > 0 ? (array[0] as T) : null;
+}
+    `,
+    `
+function first<T extends string | null>(array: ArrayLike<T>): T | null {
```

Comment:
Just for completeness, could you also add tests for `string | undefined` and `string | null | undefined`? Nullability / flag checking can be tricky & gets busted over time a lot.